Dev guide: review read data (#8666)
pingcap/docs: develop/dev-guide-use-views.md

---
title: Views
summary: Introduce how to use views in TiDB.
---

# Views

This document introduces how to use views in TiDB.

## Overview

TiDB supports views. A view acts as a virtual table, whose schema is defined by the `SELECT` statement that creates the view.

> - Exposing only safe fields and data to users to ensure the security of sensitive fields and data stored in the underlying table.
> - Defining complex queries that frequently appear as views to make complex queries easier and more convenient.

Reviewer: ```suggestion
- You can create views to expose only safe fields and data to users, which ensures the security of sensitive fields and data in the underlying tables.
- You can create views for complex queries that are frequently used to make complex queries easier and more convenient.
```

## Create a view

In TiDB, a complex query can be defined as a view with the `CREATE VIEW` statement. The syntax is as follows:

```sql
CREATE VIEW view_name AS query;
```

Note that you cannot create a view with the same name as an existing view or table.

For example, the [multi-table join query](/develop/dev-guide-join-tables.md) gets a list of books with average ratings by joining the `books` table and the `ratings` table through a `JOIN` statement.

> For the convenience of subsequent queries, the following statement defines the query statement as a view:

Reviewer: ```suggestion
For the convenience of subsequent queries, you can define the query as a view using the following statement:
```

GROUP BY b.id;
```

## Query views

> Once the view is created, you can use the `SELECT` statement to query the view just like a normal table.

Reviewer: ```suggestion
Once a view is created, you can use the `SELECT` statement to query the view just like a normal table.
```

## Limitation

> For more information about the limitations of views, see [Limitations of Views](/views.md#limitations).

Reviewer: ```suggestion
For limitations of views in TiDB, see [Limitations of Views](/views.md#limitations).
```
